mteb/tasks/InstructionRetrieval/ara/WikipediaInstructionRetrieval.py

```diff
@@ -0,0 +1,124 @@
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks.AbsTaskInstructionRetrieval import AbsTaskInstructionRetrieval
+
+_EVAL_SPLIT = "dev"
+
+_LANGUAGES = {
+    "ar": ["ara-Arab"],
+}
+
+
+def _load_wikipedia_data(
+    path: str, langs: list, splits: str, cache_dir: str = None, revision: str = None
+):
+    corpus = {lang: {split: None for split in splits} for lang in langs}
+    queries = {lang: {split: None for split in splits} for lang in langs}
+    relevant_docs = {lang: {split: None for split in splits} for lang in langs}
+
+    split = _EVAL_SPLIT
+
+    for lang in langs:
+        # Load corpus data (Can be several millions for languages)
+        corpus_identifier = f"corpus-{lang}"
+        corpus_data = datasets.load_dataset(
+            path,
+            corpus_identifier,
+            cache_dir=cache_dir,
+            revision=revision,
+            trust_remote_code=True,
+        )
+        corpus[lang][split] = {}
+        for row in corpus_data["corpus"]:
+            docid = row["docid"]
+            doc_title = row["title"]
+            doc_text = row["text"]
+            corpus[lang][split][docid] = {"title": doc_title, "text": doc_text}
+
+        # Load queries data
+        queries_identifier = f"queries-{lang}"
+        queries_data = datasets.load_dataset(
+            path,
+            queries_identifier,
+            cache_dir=cache_dir,
+            revision=revision,
+            trust_remote_code=True,
+        )
+        queries[lang][split] = {}
+        for row in queries_data["queries"]:
+            query_id = row["query_id"]
+            query_text = row["query"]
+            queries[lang][split][query_id] = query_text
+
+        # Load relevant documents data
+        qrels_identifier = f"{lang}"
+        qrels_data = datasets.load_dataset(
+            path,
+            qrels_identifier,
+            cache_dir=cache_dir,
+            revision=revision,
+            trust_remote_code=True,
+        )
+        relevant_docs[lang][split] = {}
+        for row in qrels_data[split]:
+            query_id = row["query_id"]
+            doc_id = row["docid"]
+            score = row["score"]
+            if query_id not in relevant_docs[lang][split]:
+                relevant_docs[lang][split][query_id] = {}
+            relevant_docs[lang][split][query_id][doc_id] = score
+
+    corpus = datasets.DatasetDict(corpus)
+    queries = datasets.DatasetDict(queries)
+    relevant_docs = datasets.DatasetDict(relevant_docs)
+
+    return corpus, queries, relevant_docs
+
+
+
+class WikipediaInstructionRetrieval(AbsTaskInstructionRetrieval):
+    metadata = TaskMetadata(
+        name="WikipediaInstructionRetrieval",
+        description="This dataset contains a pre-processed version from Wikipedia suitable for semantic search.",
+        reference="https://huggingface.co/datasets/Cohere/wikipedia-22-12-ar-embeddings",
+        dataset={
+            "path": "Cohere/wikipedia-22-12-ar-embeddings",
```

Comment:
I'm not sure how this dataset can be used for retrieval. It doesn't contain queries according to what's on HF, I only see pargraph ids, doc ids and text content but no queries.
I think we're missing the queries here 🤔

Reply:
There are these two datasets although it's part of miracl, which we already handle: 
![image](https://github.com/embeddings-benchmark/mteb/assets/33312980/154550b8-c81f-4373-ab07-083a3dbbcf32)